contributing.md

```diff
@@ -62,30 +149,100 @@ npm install
 
-To install the `pre-commit` hook, do `bash vendor/xwp/wp-dev-lib/scripts/install-pre-commit-hook.sh`.
+## Developing the Plugin
 
-Note that pull requests will be checked against [WordPress-Coding-Standards](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards) with PHPCS, and for JavaScript linting is done with ESLint and (for now) JSCS and JSHint.
+Whether you use the pre-existing local environment or a custom one, any PHP code changes will be directly visible during development.
 
-## Modifying JavaScript for Blocks
+However, for JavaScript this involves a build process. To watch for any JavaScript file changes and re-build it when needed, you can run the following command:
 
-To edit JavaScript code which is built/complied, run `npm run dev` to watch the files which Webpack will build. These complied files are excluded from version control but they are included in the release packages.
+```bash
+npm run dev
+```
 
-## Creating a Plugin Build
+This way you will get a development build of the JavaScript, which makes debugging easier.
 
-To create a build of the plugin for installing in WordPress as a ZIP package, do:
+To get a production build, run:
+
+```bash
+npm run build:js
+```
+
+### Creating a Plugin Build
+
+To create a build of the plugin for installing in WordPress as a ZIP package, run:
 
 ```bash
-composer install # (if you haven't done so yet)
-npm install # (if you haven't done so yet)
 npm run build
 ```
 
-This will create an `amp.zip` in the plugin directory which you can install. The contents of this ZIP are also located in the `build` directory which you can `rsync` somewhere as well.
+This will create an `amp.zip` in the plugin directory which you can install. The contents of this ZIP are also located in the `build` directory which you can `rsync` somewhere as well if needed.
+
+### Writing Tests
+
+#### PHP Unit Tests
+
+The AMP plugin uses the [PHPUnit](https://phpunit.de/) testing framework to write unit and integration tests for the PHP part.
+
+To run the full test suite, you can use the following command:
+
+```bash
+npm run test:php
+```
+
+You can also just run test for a specific function or class by using something like this:
+
+```bash
+npm run test:php -- --filter=AMP_Theme_Support
+```
+
+See `npm run test:php:help` to see all the possible options.
+
+#### JavaScript Unit Tests
+
+[Jest](https://jestjs.io/) is used as the JavaScript unit testing framework.
+
+To run the full test suite, you can use the following command:
+
+```bash
+npm run test:js
+```
+
+You can also watch for any file changes and only run tests that failed or have been modified:
+
+```bash
+npm run test:js:watch
+```
+
+See `npm run test:js:help` to get a list of additional options that can be passed to the test runner. 
+
+#### End-to-End Tests
+
+This project leverages the local Docker-based environment to facilitate end-to-end (e2e) testing using Puppeteer.
```

Comment:
If someone's not using that environment, it might be possible to run something like this:

```bash
npm run test:e2e -- --wordpress-base-url=https://amp-wp.local
```